docs/design-documents/20240430-schema-support.md

```diff
@@ -0,0 +1,161 @@
+# Schema support
+
+## The problem
+
+A Conduit user would like to start using a destination connector **without**
+manually setting up
+the [target collections](https://conduit.io/docs/introduction/vocabulary).
+
+To do so, the following is required:
+
+* collection name
+* collection metadata
+* schema
+
+This design document focuses on the schema support in Conduit in the context of
+the above.
+
+## Requirements
+
+1. Records **should not** carry the full schema. 
+
+   Reason: If a record would carry the whole schema, that would increase the
+   record size a lot.
+2. Sources and destinations need to be able to work with multiple schemas.
+   
+   Reason: Multiple collections support.
+3. A schema should be accessible across pipelines and Conduit instances.
+
+   Reason: Pipelines might work together to stream data from A to B, through an
+   intermediary C. For example, PostgreSQL to Kafka, and then Kafka to
+   Snowflake. Also, in the future, pipelines might run in isolation on different
+   instances.
+4. It should be possible for a schema to evolve.
+5. A source connector should be able to register a schema.
+6. A destination connector should be able to fetch a specific schema.
+7. A destination connector needs a way to know that a schema changed.
+8. The Connector SDK should cache the schemas.
+
+   Reason: Standalone connectors and Conduit communicate via gRPC. To avoid the
+   cost of repeatedly fetching the same schema many times (especially over
+   gRPC), schemas should be cached by the SDK.
+
+## Schema format
+
+A destination connector should work with one schema format only, regardless of
+the underlying or intermediary schema formats used. This makes it easy for a
+connector developer to write code, since it doesn't require handling of
+potentially multiple schema types.
+
+A schema consists of following:
+* reference: a string that uniquely identifies a schema in Conduit
+* list of fields, where each field is described with following:
+  * name
+  * type
+  * optional (boolean value)
+  * default value
+
+The following types are supported:
+* Primitive:
+  * boolean
+  * integers: 8, 16, 32, 64-bit 
+  * float: single precision (32-bit) and double precision (64-bit) IEEE 754 floating-point number
+  * bytes
+  * string
+* Complex:
```

Comment:
Do want to add "time" as a support type? Or are these just native golang/protobuf types?

Reply:
No special reason why I left it out, I'll add it.:)

Reply:
There are probably going to be more, but Time seems to be the one persistent across most payloads